### ISSUE TITLE
Failures in connection and connect behavior

### ISSUE THREAD

**Issue by rodmaz** [aws/aws-iot-device-sdk-js-v2#240]
### Describe the bug

It is known that `aws-crt` uses Happy Eyeballs to resolve DNS entries from AWS IoT ATS. In our case, it resolves up to 16 IP address (both IPv4 and IPv6).
Whenever we call the `MqttClientConnection.connect` and device fails to connect to any IPv6 address, `MqttClientConnection.connect` throws an exception and the NodeJS app finishes.

Following what is described [here](https://github.com/aws/aws-iot-device-sdk-js-v2/issues/176#issuecomment-909728608) by @TingDaoK , we should catch the error so the app does not die. That does not solve all the problems though. the documentation is not clear who is responsible to handle the errors in the address resolution phase.

### Expected Behavior

[`MqttClientConnection.connect`](https://aws.github.io/aws-iot-device-sdk-js-v2/classes/mqtt.mqttclientconnection.html#connect) method expected to try all IP addresses and only throw an exception IF device is unable to connect to ANY of the IPs resolved.

### Current Behavior

Whenever we call the `MqttClientConnection.connect` and device fails to connect to any IPv6 address, `MqttClientConnection.connect` throws an exception and the NodeJS app finishes.

if we try to use the `reconnect` (deprecated) method, the following errors are logged:

```
an error occurred while connecting to IoT Core; Error: Error: aws-c-common: AWS_ERROR_UNSUPPORTED_OPERATION, Unsupported operation.
an error occurred while trying to connect to AWS IoT ATS; Error: aws-c-common: AWS_ERROR_UNSUPPORTED_OPERATION, Unsupported operation.
```

### Reproduction Steps

```
  const client = new mqtt.MqttClient(client_bootstrap);
    const connection = client.new_connection(config)
      .on('connect', (session_present) => {
      console.info(`connected to AWS IoT Core ATS endpoint; session_present: ${session_present}`);
    })
      .on('disconnect', () => {
      console.info('disconnected from AWS IoT Core ATS endpoint');
    })
      .on('error', (error) => {
      console.error(`an error occurred while connecting to IoT Core; ${error}`);
    })
      .on('interrupt', (error) => {
      console.error(`connection to IoT Core ATS dropped unexpectedly; ${error}`);
    });

    const identity = new iotidentity.IotIdentityClient(connection);

    try {
      await connect();
    } catch (err) {
      console.error(`an error occurred while trying to connect to AWS IoT ATS; ${err}`);
    }
```

### Possible Solution

Is the developer responsibility to retry connections when the library is resolving IP addresses in `aws-crt`?

### SDK version used

1.8.1

### Environment details (OS name and version, etc.)

Linux, Debian, Raspberry Pi

**Comment by rodmaz** [comment]
This is the log collected in `debug` level:

```
[INFO] [2022-04-20T01:58:41Z] [0000007f849310b0] [dns] - id=0x19313750: Initializing default host resolver with 64 max host entries.
[INFO] [2022-04-20T01:58:41Z] [0000007f849310b0] [channel-bootstrap] - id=0x1922c7e0: Initializing client bootstrap with event-loop group 0x192ddd90
[DEBUG] [2022-04-20T01:58:41Z] [0000007f849310b0] [tls-handler] - ctx: Certificate and key have been set, setting them up now.
[DEBUG] [2022-04-20T01:58:41Z] [0000007f849310b0] [tls-handler] - ctx: Setting ALPN list x-amzn-mqtt-ca
[DEBUG] [2022-04-20T01:58:41Z] [0000007f849310b0] [mqtt-client] - client=0x193990e0: Initalizing MQTT client
[DEBUG] [2022-04-20T01:58:41Z] [0000007f849310b0] [mqtt-client] - id=0x192a4120: Creating new connection
[DEBUG] [2022-04-20T01:58:41Z] [0000007f849310b0] [mqtt-topic-tree] - tree=0x192a4388: Creating new topic tree
[DEBUG] [2022-04-20T01:58:41Z] [0000007f849310b0] [mqtt-client] - id=0x192a4120: Begin connecting process, switch state to CONNECTING.
[INFO] [2022-04-20T01:58:41Z] [0000007f849310b0] [mqtt-client] - id=0x192a4120: using ping timeout of 60000000000 ns
[DEBUG] [2022-04-20T01:58:41Z] [0000007f849310b0] [dns] - id=0x19313750: Host resolution requested for xxxxxxxxxxxxxxxxx-ats.iot.us-east-1.amazonaws.com
[DEBUG] [2022-04-20T01:58:41Z] [0000007f849310b0] [dns] - id=0x19313750: No cached entries found for xxxxxxxxxxxxxxxxx-ats.iot.us-east-1.amazonaws.com starting new resolver thread.
[DEBUG] [2022-04-20T01:58:41Z] [0000007f80f7f1a0] [dns] - static: resolving host xxxxxxxxxxxxxxxxx-ats.iot.us-east-1.amazonaws.com
[DEBUG] [2022-04-20T01:58:41Z] [0000007f80f7f1a0] [dns] - static: resolved record: 54.152.141.112
[DEBUG] [2022-04-20T01:58:41Z] [0000007f80f7f1a0] [dns] - static: resolved record: 35.170.227.73
[DEBUG] [2022-04-20T01:58:41Z] [0000007f80f7f1a0] [dns] - static: resolved record: 34.192.235.16
[DEBUG] [2022-04-20T01:58:41Z] [0000007f80f7f1a0] [dns] - static: resolved record: 34.200.122.79
[DEBUG] [2022-04-20T01:58:41Z] [0000007f80f7f1a0] [dns] - static: resolved record: 3.232.146.134
[DEBUG] [2022-04-20T01:58:41Z] [0000007f80f7f1a0] [dns] - static: resolved record: 34.237.102.244
[DEBUG] [2022-04-20T01:58:41Z] [0000007f80f7f1a0] [dns] - static: resolved record: 54.164.24.228
[DEBUG] [2022-04-20T01:58:41Z] [0000007f80f7f1a0] [dns] - static: resolved record: 54.162.191.159
[DEBUG] [2022-04-20T01:58:41Z] [0000007f80f7f1a0] [dns] - static: resolved record: 2406:da00:ff00::3655:f191
[DEBUG] [2022-04-20T01:58:41Z] [0000007f80f7f1a0] [dns] - static: resolved record: 2406:da00:ff00::36cc:5567
[DEBUG] [2022-04-20T01:58:41Z] [0000007f80f7f1a0] [dns] - static: resolved record: 2406:da00:ff00::365a:b04a
[DEBUG] [2022-04-20T01:58:41Z] [0000007f80f7f1a0] [dns] - static: resolved record: 2406:da00:ff00::22c1:401e
[DEBUG] [2022-04-20T01:58:41Z] [0000007f80f7f1a0] [dns] - static: resolved record: 2406:da00:ff00::36a0:6926
[DEBUG] [2022-04-20T01:58:41Z] [0000007f80f7f1a0] [dns] - static: resolved record: 2406:da00:ff00::3658:bd6
[DEBUG] [2022-04-20T01:58:41Z] [0000007f80f7f1a0] [dns] - static: resolved record: 2406:da00:ff00::22ed:62b6
[DEBUG] [2022-04-20T01:58:41Z] [0000007f80f7f1a0] [dns] - static: resolved record: 2406:da00:ff00::3ea:aba1
[DEBUG] [2022-04-20T01:58:41Z] [0000007f80f7f1a0] [dns] - static, resolving host xxxxxxxxxxxxxxxxx-ats.iot.us-east-1.amazonaws.com successful, returned 16 addresses
[DEBUG] [2022-04-20T01:58:41Z] [0000007f80f7f1a0] [dns] - static: new address resolved 54.152.141.112 for host xxxxxxxxxxxxxxxxx-ats.iot.us-east-1.amazonaws.com caching
[DEBUG] [2022-04-20T01:58:41Z] [0000007f80f7f1a0] [dns] - static: new address resolved 35.170.227.73 for host xxxxxxxxxxxxxxxxx-ats.iot.us-east-1.amazonaws.com caching
[DEBUG] [2022-04-20T01:58:41Z] [0000007f80f7f1a0] [dns] - static: new address resolved 34.192.235.16 for host xxxxxxxxxxxxxxxxx-ats.iot.us-east-1.amazonaws.com caching
[DEBUG] [2022-04-20T01:58:41Z] [0000007f80f7f1a0] [dns] - static: new address resolved 34.200.122.79 for host xxxxxxxxxxxxxxxxx-ats.iot.us-east-1.amazonaws.com caching
[DEBUG] [2022-04-20T01:58:41Z] [0000007f80f7f1a0] [dns] - static: new address resolved 3.232.146.134 for host xxxxxxxxxxxxxxxxx-ats.iot.us-east-1.amazonaws.com caching
[DEBUG] [2022-04-20T01:58:41Z] [0000007f80f7f1a0] [dns] - static: new address resolved 34.237.102.244 for host xxxxxxxxxxxxxxxxx-ats.iot.us-east-1.amazonaws.com caching
[DEBUG] [2022-04-20T01:58:41Z] [0000007f80f7f1a0] [dns] - static: new address resolved 54.164.24.228 for host xxxxxxxxxxxxxxxxx-ats.iot.us-east-1.amazonaws.com caching
[DEBUG] [2022-04-20T01:58:41Z] [0000007f80f7f1a0] [dns] - static: new address resolved 54.162.191.159 for host xxxxxxxxxxxxxxxxx-ats.iot.us-east-1.amazonaws.com caching
[DEBUG] [2022-04-20T01:58:41Z] [0000007f80f7f1a0] [dns] - static: new address resolved 2406:da00:ff00::3655:f191 for host xxxxxxxxxxxxxxxxx-ats.iot.us-east-1.amazonaws.com caching
[DEBUG] [2022-04-20T01:58:41Z] [0000007f80f7f1a0] [dns] - static: new address resolved 2406:da00:ff00::36cc:5567 for host xxxxxxxxxxxxxxxxx-ats.iot.us-east-1.amazonaws.com caching
[DEBUG] [2022-04-20T01:58:41Z] [0000007f80f7f1a0] [dns] - static: new address resolved 2406:da00:ff00::365a:b04a for host xxxxxxxxxxxxxxxxx-ats.iot.us-east-1.amazonaws.com caching
[DEBUG] [2022-04-20T01:58:41Z] [0000007f80f7f1a0] [dns] - static: new address resolved 2406:da00:ff00::22c1:401e for host xxxxxxxxxxxxxxxxx-ats.iot.us-east-1.amazonaws.com caching
[DEBUG] [2022-04-20T01:58:41Z] [0000007f80f7f1a0] [dns] - static: new address resolved 2406:da00:ff00::36a0:6926 for host xxxxxxxxxxxxxxxxx-ats.iot.us-east-1.amazonaws.com caching
[DEBUG] [2022-04-20T01:58:41Z] [0000007f80f7f1a0] [dns] - static: new address resolved 2406:da00:ff00::3658:bd6 for host xxxxxxxxxxxxxxxxx-ats.iot.us-east-1.amazonaws.com caching
[DEBUG] [2022-04-20T01:58:41Z] [0000007f80f7f1a0] [dns] - static: new address resolved 2406:da00:ff00::22ed:62b6 for host xxxxxxxxxxxxxxxxx-ats.iot.us-east-1.amazonaws.com caching
[DEBUG] [2022-04-20T01:58:41Z] [0000007f80f7f1a0] [dns] - static: new address resolved 2406:da00:ff00::3ea:aba1 for host xxxxxxxxxxxxxxxxx-ats.iot.us-east-1.amazonaws.com caching
[DEBUG] [2022-04-20T01:58:41Z] [0000007f80f7f1a0] [dns] - static, invoking resolution callback for host xxxxxxxxxxxxxxxxx-ats.iot.us-east-1.amazonaws.com with 2 addresses
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [task-scheduler] - id=0x7f68004810: Scheduling attempt_connection task for immediate execution
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [task-scheduler] - id=0x7f68005da0: Scheduling attempt_connection task for immediate execution
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [task-scheduler] - id=0x7f68004810: Running attempt_connection task with <Running> status
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [socket] - id=0x7f64000ca0 fd=25: initializing with domain 1 and type 0
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [socket] - id=0x7f64000ca0 fd=25: setting socket options to: keep-alive 0, keep idle 0, keep-alive interval 0, keep-alive probe count 0.
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [socket] - id=0x7f64000ca0 fd=25: beginning connect.
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [socket] - id=0x7f64000ca0 fd=25: connecting to endpoint 2406:da00:ff00::3655:f191:8883.
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [task-scheduler] - id=0x7f640011e0: Scheduling (null) task for future execution at time 28275318566997
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [task-scheduler] - id=0x7f68005da0: Running attempt_connection task with <Running> status
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [socket] - id=0x7f640013f0 fd=26: initializing with domain 0 and type 0
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [socket] - id=0x7f640013f0 fd=26: setting socket options to: keep-alive 0, keep idle 0, keep-alive interval 0, keep-alive probe count 0.
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [socket] - id=0x7f640013f0 fd=26: beginning connect.
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [socket] - id=0x7f640013f0 fd=26: connecting to endpoint 54.152.141.112:8883.
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [task-scheduler] - id=0x7f64001610: Scheduling (null) task for future execution at time 28275319288872
[ERROR] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [socket] - id=0x7f64000ca0 fd=25: connection failure
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [channel-bootstrap] - id=0x1922c7e0: client connection on socket 0x7f64000ca0 completed with error 1049.
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [channel-bootstrap] - id=0x1922c7e0: recording bad address 2406:da00:ff00::3655:f191.
[INFO] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [dns] - id=0x19313750: recording failure for record 2406:da00:ff00::3655:f191 for xxxxxxxxxxxxxxxxx-ats.iot.us-east-1.amazonaws.com, moving to bad list
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [socket] - id=0x7f64000ca0 fd=25: closing
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [task-scheduler] - id=0x7f64001260: Scheduling epoll_event_loop_unsubscribe_cleanup task for immediate execution
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [task-scheduler] - id=0x7f64001260: Running epoll_event_loop_unsubscribe_cleanup task with <Running> status
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [task-scheduler] - id=0x7f64001690: Scheduling epoll_event_loop_unsubscribe_cleanup task for immediate execution
[INFO] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [socket] - id=0x7f640013f0 fd=26: connection success
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [socket] - id=0x7f640013f0 fd=26: local endpoint 192.168.0.116:64658
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [socket] - id=0x7f640013f0 fd=26: assigning to event loop 0x192eaa70
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [channel-bootstrap] - id=0x1922c7e0: client connection on socket 0x7f640013f0 completed with error 0.
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [channel] - id=0x7f64001a00: Beginning creation and setup of new channel.
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [task-scheduler] - id=0x7f64001d30: Scheduling on_channel_setup_complete task for immediate execution
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [task-scheduler] - id=0x7f64001690: Running epoll_event_loop_unsubscribe_cleanup task with <Running> status
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [task-scheduler] - id=0x7f64001d30: Running on_channel_setup_complete task with <Running> status
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [channel] - id=0x7f64001a00: setup complete, notifying caller.
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [channel] - id=0x7f64001a00: no message pool is currently stored in the event-loop local storage, adding 0x7f64001d80 with max message size 16384, message count 4, with 4 small blocks of 128 bytes.
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [channel-bootstrap] - id=0x1922c7e0: channel 0x7f64001a00 setup succeeded: bootstrapping.
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [socket-handler] - id=0x7f64012530: Socket handler created with max_read_size of 16384
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [task-scheduler] - id=0x7f64012690: Scheduling tls_timeout task for future execution at time 28225459915799
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [task-scheduler] - id=0x7f64001590: Scheduling socket_written_task task for immediate execution
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [task-scheduler] - id=0x7f64001590: Running socket_written_task task with <Running> status
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [task-scheduler] - id=0x7f64001590: Scheduling socket_written_task task for immediate execution
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [task-scheduler] - id=0x7f64001590: Running socket_written_task task with <Running> status
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [tls-handler] - id=0x7f64012660: Remote server name is xxxxxxxxxxxxxxxxx-ats.iot.us-east-1.amazonaws.com
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [channel-bootstrap] - id=0x1922c7e0: tls negotiation result 0 on channel 0x7f64001a00
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [mqtt-client] - id=0x192a4120: Connection successfully opened, sending CONNECT packet
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [task-scheduler] - id=0x7f64001d10: Scheduling mqtt_connack_timeout task for future execution at time 28275931020174
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [mqtt-client] - id=0x192a4120: Adding username ?SDK=NodeJSv2&Version=1.12.0 to connection
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [task-scheduler] - id=0x7f64001590: Scheduling socket_written_task task for immediate execution
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [task-scheduler] - id=0x7f64001590: Running socket_written_task task with <Running> status
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [tls-handler] - id=0x7f64012660: Alert code 0
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [task-scheduler] - id=0x7f64001c08: Scheduling channel_shutdown task for immediate execution
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [task-scheduler] - id=0x7f64001c08: Running channel_shutdown task with <Running> status
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [channel] - id=0x7f64001a00: beginning shutdown process
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [channel] - id=0x7f64001a00: handler 0x7f64012530 shutdown in read dir completed.
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [tls-handler] - id=0x7f64012660: Shutting down read direction with error code 0
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [channel] - id=0x7f64001a00: handler 0x7f64012660 shutdown in read dir completed.
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [channel] - id=0x7f64001a00: handler 0x192a4148 shutdown in read dir completed.
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [task-scheduler] - id=0x7f64001a28: Scheduling (null) task for immediate execution
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [task-scheduler] - id=0x7f64001a28: Running (null) task with <Running> status
[INFO] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [mqtt-client] - id=0x192a4120: sending disconnect message as part of graceful shutdown.
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [mqtt-client] - id=0x192a4120: failed to send courteous disconnect io message
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [channel] - id=0x7f64001a00: handler 0x192a4148 shutdown in write dir completed.
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [tls-handler] - id=0x7f64012660: Scheduling delayed write direction shutdown
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [task-scheduler] - id=0x7f64012830: Scheduling s2n_delayed_shutdown task for future execution at time 28216085255121
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [task-scheduler] - id=0x7f64012830: Running s2n_delayed_shutdown task with <Running> status
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [tls-handler] - id=0x7f64012660: Delayed shut down in write direction
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [task-scheduler] - id=0x7f64001590: Scheduling socket_written_task task for immediate execution
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [channel] - id=0x7f64001a00: handler 0x7f64012660 shutdown in write dir completed.
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [socket] - id=0x7f640013f0 fd=26: closing
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [task-scheduler] - id=0x7f64001260: Scheduling epoll_event_loop_unsubscribe_cleanup task for immediate execution
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [task-scheduler] - id=0x7f64001590: Running socket_written_task task with <Canceled> status
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [task-scheduler] - id=0x7f640125d0: Scheduling socket_handler_close task for immediate execution
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [task-scheduler] - id=0x7f64001260: Running epoll_event_loop_unsubscribe_cleanup task with <Running> status
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [task-scheduler] - id=0x7f640125d0: Running socket_handler_close task with <Running> status
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [channel] - id=0x7f64001a00: handler 0x7f64012530 shutdown in write dir completed.
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [task-scheduler] - id=0x7f64001a28: Scheduling (null) task for immediate execution
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [task-scheduler] - id=0x7f64001a28: Running (null) task with <Running> status
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [channel] - id=0x7f64001a00: during shutdown, canceling task 0x7f64012690
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [task-scheduler] - id=0x7f64012690: Running tls_timeout task with <Canceled> status
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [channel] - id=0x7f64001a00: during shutdown, canceling task 0x7f64001d10
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [task-scheduler] - id=0x7f64001d10: Running mqtt_connack_timeout task with <Canceled> status
[DEBUG] [2022-04-20T01:58:41Z] [0000007f8180f1a0] [channel-bootstrap] - id=0x1922c7e0: channel 0x7f64001a00 shutdown with error 0.
an error occurred while connecting to IoT Core; Error: Failed to connect: libaws-c-mqtt: AWS_ERROR_MQTT_UNEXPECTED_HANGUP, The connection was closed unexpectedly.
an error occurred while trying to connect to AWS IoT ATS; Failed to connect: libaws-c-mqtt: AWS_ERROR_MQTT_UNEXPECTED_HANGUP, The connection was closed unexpectedly.
```

**Comment by rodmaz** [comment]
That's it! `aws-crt` was logging failure due to IPv6 and we failed to check the problem w/ certificates.
